src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -566,7 +559,17 @@ protected IEnumerable<string> GetAllValidProjectNames()
         /// <summary>
         /// Get list of packages from the remote package source. Used for Get-Package -ListAvailable.
         /// </summary>
-        protected IEnumerable<IPackageSearchMetadata> GetPackagesFromRemoteSource(string searchString, bool includePrerelease)
+        /// <param name="searchString">The search string to use for filtering.</param>
+        /// <param name="includePrerelease">Whether or not to include prerelease packages in the results.</param>
+        /// <param name="handleException">
+        /// An action for handling errors during the enumeration of the returned results. The
+        /// parameter is the error message. This action is very called by multiple threads at once.
+        /// </param>
+        /// <returns>The lazy sequence of package search metadata.</returns>
+        protected IEnumerable<IPackageSearchMetadata> GetPackagesFromRemoteSource(
+            string searchString,
+            bool includePrerelease,
+            Action<string> handleError)
```

Comment:
the same way you can pass `handleError` predicate to `PackageFeedEnumerator.Enumerate` and further to Enumerator itself